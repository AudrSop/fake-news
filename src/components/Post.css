.Post {
  border: 1px solid lightgray;
  border-radius: 3px;
  background-color: white;
  padding: .8rem;
  margin: .8rem;
}

.Post-title {
  font-weight: 700;
  font-size: 1.2rem;
  font-weight: 600;
}

.Post-body {
  color: #1D2129;
  font-weight: 500;
  text-align: left;
}