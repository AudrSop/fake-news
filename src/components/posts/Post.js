import React, { Component } from 'react';
import { Loader } from '..';
import { api } from '../../services';
import './Post.css';

class Post extends Component {
  state = {
    post: null,
    shoudCancel: false
  }

  componentDidMount() {
    this.loadPost();
  }

  componentWillUnmount() {
    this.shoudCancel = true;
  }

  loadPost = async () => {
    const { match: { params } } = this.props;
    const post = await api.Posts.findOne(params.id);
    if (!this.shoudCancel) {
      this.setState({ post });
    }
  }

  render() {
    const { post } = this.state;
    if (!post) return <Loader />;
    return (
      <article className="Post">
        <h1 className="Post-title">{post.title}</h1>
        <p className="Post-body">{post.body}</p>
      </article>
    );
  }
}

export default Post;
