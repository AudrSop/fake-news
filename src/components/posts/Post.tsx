import React, { Component } from 'react';
import { Loader } from 'src/components';
import { PostEntity } from 'src/model';
import { api } from 'src/services';
import './Post.css';

interface PostProps {
  match: any;
}

interface State {
  post: PostEntity;
}

class Post extends Component<PostProps, State> {
  componentDidMount() {
    this.loadPost();
  }

  loadPost = async () => {
    const { match: { params } } = this.props;
    try {
      const post = await api.Posts.findOne(params.id);
      this.setState({ post });
    } catch(e) {
      // tslint:disable-next-line
      console.error(e);
    }
  }

  render() {
    const { post } = this.state;

    if (!post) {
      return <Loader />;
    }

    return (
      <article className="Post">
        <h1 className="Post-title">{post.title}</h1>
        <p className="Post-body">{post.body}</p>
      </article>
    );
  }
}

export default Post;
