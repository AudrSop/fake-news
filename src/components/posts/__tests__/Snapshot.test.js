import React from 'react';
// import renderer from 'react-test-renderer';
import Post from '../Post';
import { MirrorRouter } from 'react-router-dom';
import { mount, shallow } from 'enzyme';

it('renders correctly', () => {
  const post = {id: 1, title: 'One', body: 'Text'};

  const wrapper = shallow(<Post post={post}></Post>);
  expect(wrapper).toMatchSnapshot();
});