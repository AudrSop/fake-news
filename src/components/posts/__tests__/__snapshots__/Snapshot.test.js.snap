// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Post
    post={
        Object {
            "body": "Text",
            "id": 1,
            "title": "One",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/posts/1"
>
          <h1
                    className="Post-title"
          >
                    One
          </h1>
</Link>,
        <p
          className="Post-body"
>
          Text
</p>,
      ],
      "className": "Post",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="Post-title"
>
            One
</h1>,
          "replace": false,
          "to": "/posts/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "One",
            "className": "Post-title",
          },
          "ref": null,
          "rendered": "One",
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Text",
          "className": "Post-body",
        },
        "ref": null,
        "rendered": "Text",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/posts/1"
>
            <h1
                        className="Post-title"
            >
                        One
            </h1>
</Link>,
          <p
            className="Post-body"
>
            Text
</p>,
        ],
        "className": "Post",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1
              className="Post-title"
>
              One
</h1>,
            "replace": false,
            "to": "/posts/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "One",
              "className": "Post-title",
            },
            "ref": null,
            "rendered": "One",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text",
            "className": "Post-body",
          },
          "ref": null,
          "rendered": "Text",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
